docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
+
+.. code-block:: python
+
+    from aiida.engine import run
+    from aiida.manage.caching import enable_caching
+    with enable_caching(identifier='aiida.calculations:templatereplacer'):
+       run(...)
+
+.. warning::
+
+    This affects only the current python interpreter and won't change the behavior of the daemon workers.
+    This means that this technique is only useful when using :py:class:`~aiida.engine.run`, and **not** with :py:class:`~aiida.engine.submit`.
+
+If you suspect a node is being reused in error (e.g. during development), you can also manually *prevent* a specific node from being reused:
+
+1. Load one of the nodes you suspect to be a clone.
+   Check that :meth:`~aiida.orm.nodes.Node.get_cache_source` returns a UUID.
+   If it returns `None`, the node was not cloned.
+2. Clear the hashes of all nodes that are considered identical to this node:
+
+    .. code:: python
+
+        for node in node.get_all_same_nodes():
+            node.clear_hash()
+3. Run your calculation again.
+   The node in question should no longer be reused.
+
+
+.. _how-to:run-codes:caching:limitations:
+
+Limitations
+-----------
+
+#. Workflow nodes are not cached.
+   In the current design this follows from the requirement that the provenance graph be independent of whether caching is enabled or not:
+
+   * **Calculation nodes:** Calculation nodes can have data inputs and create new data nodes as outputs.
+     In order to make it look as if a cloned calculation produced its own outputs, the output nodes are copied and linked as well.
+   * **Workflow nodes:** Workflows differ from calculations in that they can *return* an input node or an output node created by a calculation.
+     Since caching does not care about the *identity* of input nodes but only their *content*, it is not straightforward to figure out which node to return in a cached workflow.
+
+   For the moment, this limitation is acceptable since the runtime of AiiDA WorkChains is usually dominated by expensive calculations, which are covered by the current caching mechanism.
+
+#. The caching mechanism for calculations *should* trigger only when the inputs and the calculation to be performed are exactly the same.
+   While AiiDA's hashes include the version of the python package containing the calculation/data classes, it cannot detect cases where the underlying python code was changed without increasing the version number.
+   Another edge case would be if the parser lives in a different python package than the calculation (calculation nodes store the name of the parser used but not the version of the package containing the parser).
 
-    `#3988`_
+Finally, while caching saves unnecessary computations, it does not save disk space: The output nodes of the cached calculation are full copies of the original outputs.
```

Comment:
Is there any reason that this is shown as  separate paragraph instead of just another bullet point?

Reply:
Don't think so. Will include as bullet point